lib/auth/clt.go

```diff
@@ -199,19 +152,26 @@ type ClientConfig struct {
 // CheckAndSetDefaults checks and sets default config values
 func (c *ClientConfig) CheckAndSetDefaults() error {
```

Comment:
Seems like this is the same logic as in api/client. Can it be reused? Maybe use the same config struct as in api.Client here.

Reply:
Ideally yes, but `/lib/auth/clt.go` uses `[]utils.NetAddr` in client config, while `/api/client.go` uses `[]string` to avoid importing utils. Since we don't want to alter the currently supported api config struct, I had to duplicate the logic. If you see a way around this let me know though, I've tried already because it bothers me too.

Reply:
> If you see a way around this let me know though, I've tried already because it bothers me too.

Well, one idea is to refactor whatever's using `auth.ClientConfig` to use `[]string` as well instead of `[]utils.NetAddr`. Is that a lot of work?

Reply:
Here, https://github.com/gravitational/teleport/pull/5072#discussion_r538985979, we talked a tiny bit about this before when I tried it. I don't remember what made me revert it, I think I was worried about backwards compatibility of `NewTLSClient` in `lib/auth/clt.go`. This should be a simple change for users who upgrade to the new API, but I'm still hesitant. I'm going to commit the change so we can see it, and then revert if needed.

Reply:
Unfortunately we can't factor out `Addrs: []utils.NetAddr` completely because it'll break backwards compatibility, but we can actually embed `api.Config`, and deprecate the old `Addrs`.
```
type ClientConfig struct {
	api.Config
	// Addrs is a list of addresses to dial (deprecated)
	Addrs []utils.NetAddr
}
```